tests/UKHO.ERPFacade.API.FunctionalTests/Helpers/RoSXMLHelper.cs

```diff
@@ -0,0 +1,239 @@
+                s_attrNotMatched.Add(nameof(roSXmlField.ENDDATE));
+            if (!roSXmlField.VNAME.Equals(roSJsonFields.vesselName))
+                s_attrNotMatched.Add(nameof(roSXmlField.VNAME));
+            if (!roSXmlField.IMO.Equals(roSJsonFields.imoNumber))
+                s_attrNotMatched.Add(nameof(roSXmlField.IMO));
+            if (!roSXmlField.CALLSIGN.Equals(roSJsonFields.callSign))
+                s_attrNotMatched.Add(nameof(roSXmlField.CALLSIGN));
+            if (!roSXmlField.SHOREBASED.Equals(roSJsonFields.shoreBased))
+                s_attrNotMatched.Add(nameof(roSXmlField.SHOREBASED));
+            if (!roSXmlField.FLEET.Equals(roSJsonFields.fleetName))
+                s_attrNotMatched.Add(nameof(roSXmlField.FLEET));
+            if (!roSXmlField.USERS.Equals(roSJsonFields.numberLicenceUsers))
+                s_attrNotMatched.Add(nameof(roSXmlField.USERS));
+            if (!roSXmlField.ENDUSERID.Equals(roSJsonFields.licenseId))
+                s_attrNotMatched.Add(nameof(roSXmlField.ENDUSERID));
+            if (!roSXmlField.ECDISMANUF.Equals(roSJsonFields.upn))
+                s_attrNotMatched.Add(nameof(roSXmlField.ECDISMANUF));
+            if (!roSXmlField.LTYPE.Equals(roSJsonFields.licenceType))
+                s_attrNotMatched.Add(nameof(roSXmlField.LTYPE));
+            if (!roSXmlField.LICDUR.Equals(roSJsonFields.licenceDuration))
+                s_attrNotMatched.Add(nameof(roSXmlField.LICDUR));
+            
+
+
+            string[] fieldNames = { "LICNO","FLEET" };
+            string[] fieldNamesProduct = { "REPEAT" };
+            Z_ADDS_ROSIM_ORDERItem[] items = roSXmlField.PROD;
+            JsonInputRoSWebhookHelper.Unitsofsale[] unitofsales = roSJsonFields.unitsOfSale;
+            VerifyBlankFields(roSXmlField, fieldNames);
+            VerifyProductFields(items, unitofsales);
+            VerifyBlankProductFields(items, fieldNamesProduct);
+            if (s_attrNotMatched.Count == 0)
+            {
+                Console.WriteLine("NEWLICENCE event XML is correct");
+                return true;
+            }
+            else
+            {
+                Console.WriteLine("NEWLICENCE event XML is incorrect");
+                Console.WriteLine("Not matching attributes are:");
+                foreach (string attribute in s_attrNotMatched)
+                { Console.WriteLine(attribute); }
+                return false;
+            }
+        }
+
+        private static void VerifyBlankFields(Z_ADDS_ROSIM_ORDER roSXmlField, string[] fieldNames)
+        {
+            foreach (string field in fieldNames)
+            {
+                if (!typeof(Z_ADDS_ROSIM_ORDER).GetProperty(field).GetValue(roSXmlField, null).Equals(""))
+                    s_attrNotMatched.Add(typeof(Z_ADDS_ROSIM_ORDER).GetProperty(field).Name);
+            }
+        }
+
+        private static void VerifyProductFields(Z_ADDS_ROSIM_ORDERItem[] roSResult, JsonInputRoSWebhookHelper.Unitsofsale[] unitofsales)
```

Comment:
Model class should not be refer as helper class

Reply:
done